docs/src/index.md

```diff
@@ -525,11 +525,111 @@ To reference columns with more complicated expressions, you must wrap column ref
+In operations, it is also allowed to use `AsTable(cols)` to work with
+multiple columns at once, where the columns are grouped together in a
+`NamedTuple`. When `AsTable(cols)` appears in a operation, no
+other columns may be referenced in the block.
 
-### Multi-argument selectors 
+`AsTable` on the right-hand-side also allows the use of the special
+column selectors `Not`, `Between`, and regular expressions. As well
+as working with lists of variables programmatically. 
+
+For example, consider a collection of column names `vars`, such that
+
+```
+df = DataFrame(a = [11, 14], b = [17, 10], c = [12, 5]);
+vars = ["a", "b"];
+```
+
+To make a new column which is the sum of `vars`, write
+
+```
+julia> @rtransform df :y = sum(AsTable(vars))
+2×4 DataFrame
+ Row │ a      b      c      y     
+     │ Int64  Int64  Int64  Int64 
+─────┼────────────────────────────
+   1 │    11     17     12     28
+   2 │    14     10      5     24
+```
+
+To subset all rows where the sum is greater than `25`, write
+
+```
+julia> @rsubset df sum(AsTable(vars)) > 25
+1×3 DataFrame
+ Row │ a      b      c     
+     │ Int64  Int64  Int64 
+─────┼─────────────────────
+   1 │    11     17     12
+```
+
+To understand the how this works, recall that DataFrames.jl allows for
+`AsTable(cols)` to be a `source` in a `source => fun => dest` mini-language
+expression. As a consequence, the transformation call
+
+```
+:y = f(AsTable(cols)) 
+```
+
+becomes
+
+```
+AsTable(cols) => f => :y
+```
+
+Note that DataFrames does *not* allow `source => fun => dest` commands 
+to be of the form 
+
+```
+[AsTable(cols), :x] => f => :y
+```
+
+As a consequence, DataFramesMeta.jl does not allow any other column selectors to appear 
+inside the expression. The command
+
+```
+:y = sum(AsTable(cols)) + :d
+```
+
+will fail. 
+
+Finally, note that everyting inside `AsTable` is escaped by default. There is no ned to use `$` inside `AsTable` on the right-hand-side. For example
+
+```
+:y = first(AsTable("a"))
+```
+
+will work as expected.  
+
+
+### AsTable and `@astable`, explained
+
+At this point we have seen `AsTable` appear in three places:
+
+1. `AsTable` on the left-hand-side of transformations: `$AsTable = f(:a, :b)`
+2. The macro-flag `@astable` within the transformation. 
+3. `AsTable(cols)` on the right hand side for multi-column transformations. 
+
+The differences between the three is summarized below
+
+| Operation         | Purpose                                                                             | Example           | Notes |
+|-------------------|-------------------------------------------------------------------------------------|-------------------|-------|
+| `$AsTable` on LHS | Create multiple columns at once, whose column names are only known programmatically | <pre lang=julia>$AsTable = f(:y)</pre> | Requires escaping with `$` until deprecation period ends for unquoted column names on LHS.      |
```

Comment:
Pretty hacky way to make a reasonable little table. But it looks nice when compiled.

Reply:
It does not work, unfortunately. I will think of another way to do this.